diff --git a/index.js b/index.js
--- a/index.js
+++ b/index.js
@@ -5,7 +5,7 @@ const url = require('url');
 const caw = require('caw');
 const contentDisposition = require('content-disposition');
 const archiveType = require('archive-type');
-const decompress = require('decompress');
+const decompress = (...args) => import('decompress').then(({default: extract}) => extract(...args));
 const filenamify = require('filenamify');
 const getStream = require('get-stream');
 const got = require('got');
